### PR TITLE
feat(core): date/time locale support + tick formatting

### LINKED ISSUE #388: [LineChart]: Need localization support for the axis of time series
#### I am submitting a...

- [ ] Feature request
- [ ] Design defect
- [x] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
0.16.24
### Issue description
Set your language to French and view a LineChart graph instead of seeing the French months you see them in English

image

### PR DESCRIPTION
**Pull request by theiliad**
Also updates typescript to latest version, because of the need to use `resolveJsonModule`

Closes #388

### CODE REVIEW

File: packages/core/src/components/axes/axis.ts [hunk 1]
```diff
@@ -189,9 +190,18 @@ export class Axis extends Component {
 				break;
 		}
 
+		// Set the date/time locale
+		if (this.scaleType === ScaleTypes.TIME) {
+			const timeLocale = Tools.getProperty(options, "locale", "time");
+			if (timeLocale) {
+				timeFormatDefaultLocale(timeLocale);
```
Comment (cal-smith) [inline packages/core/src/components/axes/axis.ts]: It's kinda spooky how that just makes everything work 👻

Comment (cal-smith) [reply packages/core/src/components/axes/axis.ts]: But it does work! And that's pretty sweet

**Comment by theiliad** [comment]
Just noticed, storybooks are empty, I'll fix

**Comment by theiliad** [comment]
@cal-smith @natashadecoste good to go

**Comment by scottdickerson** [comment]
@theiliad I resolved the merge issue, any idea when this might be merged in?

**Pull request merged**